luabind/detail/object_rep.hpp

```diff
@@ -98,7 +102,11 @@ namespace luabind { namespace detail
         void operator=(object_rep const&);
 
         BOOST_STATIC_CONSTANT(std::size_t, instance_buffer_size=32);
+#ifdef BOOST_NO_CXX11_HDR_TYPE_TRAITS
         boost::aligned_storage<instance_buffer_size> m_instance_buffer;
```

Comment:
I think there was a silly mistake in this code, namely missing `::type`:

```suggestion
        boost::aligned_storage<instance_buffer_size>::type m_instance_buffer;
```

Can you check if that fixes it?